packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

```diff
@@ -517,36 +545,46 @@ export interface ApplicationListenerAttributes {
   readonly securityGroupAllowsAllOutbound?: boolean;
 }
 
-class ImportedApplicationListener extends Resource implements IApplicationListener {
+/**
+ * Props for `ImportedApplicationListenerBase`
+ */
+interface ImportedApplicationListenerBaseProps {
+  /**
+   * Arn of the listener.
+   */
+  readonly listenerArn: string;
+
+  /**
+   * Connections object.
+   */
+  readonly connections: ec2.Connections;
+}
+
+abstract class ImportedApplicationListenerBase extends Resource implements IApplicationListener {
+  /**
+   * Connections object.
+   */
   public readonly connections: ec2.Connections;
 
   /**
    * ARN of the listener
    */
   public readonly listenerArn: string;
 
-  constructor(scope: Construct, id: string, props: ApplicationListenerAttributes) {
+  constructor(scope: Construct, id: string, props: ImportedApplicationListenerBaseProps) {
     super(scope, id);
 
+    this.connections = props.connections;
     this.listenerArn = props.listenerArn;
```

Comment:
I think a cleaner approach would be to keep `connections` and `listenerArn` abstract in this base class, and just set them in the concrete classes. Saves needing to create the `ImportedApplicationListenerBaseProps` and awkward `super` calls.